session/session_test.go

```diff
@@ -0,0 +1,57 @@
+package session
+
+import (
+	toxiproxy "github.com/Shopify/toxiproxy/client"
+	"testing"
+	"time"
+)
+
+func TestReceiveEventWhenSubscribing(t *testing.T) {
+	client := toxiproxy.NewClient("http://localhost:8474")
+	proxy := client.NewProxy(&toxiproxy.Proxy{Name: "gozk_test_zookeeper", Listen: "localhost:27445", Upstream: "localhost:2181", Enabled: true})
+
+	err := proxy.Create()
+	if err != nil {
+		t.Error("Failed to create proxy: ", err)
+	}
+	defer proxy.Delete()
+
+	store, err := NewZKSession("localhost:27445", 200*time.Millisecond)
+	if err != nil {
+		t.Error("Failed to connect to Zookeeper: ", err)
+	}
+	defer store.Close()
+
+	events := make(chan ZKSessionEvent)
+	store.Subscribe(events)
+
+	go func() {
+		err := proxy.Delete()
+		if err != nil {
+			t.Error("Failed to delete proxy: ", err)
+		}
+
+		_, _, err = store.Children("/")
+		if err == nil {
+			t.Error("Expected error when listing children")
+		}
+
+		err = proxy.Create()
+		if err != nil {
+			t.Error("Failed to create proxy: ", err)
+		}
+	}()
+
+	select {
+	case event := <-events:
+		if event != SessionDisconnected {
+			t.Error("Expected to receive disconnected: ", event)
+		}
+
+		if <-events != SessionReconnected {
```

Comment:
I was thinking about that but I think it just makes the code more complex to read for a case that never came up during my testing (I wrote the tests first). I can add it if you think it's worth it.